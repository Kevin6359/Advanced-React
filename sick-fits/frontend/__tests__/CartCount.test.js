import { render, screen } from '@testing-library/react';
import wait from 'waait';
import CartCount from '../components/CartCount';

describe('<CartCount/>', () => {
    it('renders', () => {
        const {container, debug} = render(<CartCount count={1}/>);
    });
    it('updates via props', async () => {
        const {container, rerender, debug} = render(<CartCount count={6}/>);
        expect(container.textContent).toBe('6');
        rerender(<CartCount count={12}/>);
        await wait(400);
        // await screen.findByText('12')
        expect(container.textContent).toBe('12');
    });
});