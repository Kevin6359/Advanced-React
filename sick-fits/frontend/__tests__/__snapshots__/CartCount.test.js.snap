// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> matches the snapshot 1`] = `
<div>
  <div
    class="CartCount__Dot-sc-xxvp4g-0 iephzA"
  >
    15
  </div>
</div>
`;

exports[`<CartCount/> updates via props 1`] = `
<div>
  <div
    class="CartCount__Dot-sc-xxvp4g-0 iephzA"
  >
    12
  </div>
</div>
`;
